clarity/contracts/arkadiko-swap-v1-1.clar

```diff
@@ -211,8 +232,88 @@
   (ok (var-get pair-count))
 )
 
-(define-read-only (get-pairs)
-  (ok (map get-pair-contracts (var-get pairs-list)))
+(define-public (migrate-create-pair (token-x-trait <ft-trait>) (token-y-trait <ft-trait>) (swap-token-trait <swap-token>) (pair-name (string-ascii 32)) (shares-total uint))
```

Comment:
Not sure to understand why this function is being introduced, could you comment on this? What would have been the issue with using the `create-pair` method?

Reply:
We needed a migration method so we could set the `total-shares` in the pair. The `create-pair` initialises the pair and sets the total shares to zero. This allows us to create a pair, set the total shares, and then later add liquidity through it with `migrate-add-liquidity`

We preferred this code duplication over changing the arguments in the `create-pair` method since `create-pair` will be used often in the future.